File: NotionServer/TimeService.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;

namespace NotionServer
{
    public class TimeService
    {
        private DateTime _time;

        public async Task<DateTime> GetTime()
        {
            string url = "https://timeapi.io/api/Time/current/zone?timeZone=Europe/Moscow";

            using HttpClient client = new HttpClient();

            try
            {
                HttpResponseMessage message = await client.GetAsync(url);
                message.EnsureSuccessStatusCode();

                string json = await message.Content.ReadAsStringAsync();
                JsonElement data = JsonSerializer.Deserialize<JsonElement>(json);
                int year = data.GetProperty("year").GetInt32();
                int month = data.GetProperty("month").GetInt32();
                int day = data.GetProperty("day").GetInt32();
                int hour = data.GetProperty("hour").GetInt32();
                int minute = data.GetProperty("minute").GetInt32();
                int seconds = data.GetProperty("seconds").GetInt32();

                DateTime dateTime = new DateTime(year, month, day, hour, minute, seconds);

                return dateTime;
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Ошибка: {ex.Message}");
                return new DateTime();
            }
        }
    }
}

File: fastApi/services/timeService.py
import httpx
import asyncio

# current_time = None
# URL = "https://timeapi.io/api/Time/current/zone?timeZone=Europe/Moscow"

# async def startTimeService():
#     global current_time
#     async with httpx.AsyncClient() as client:
#         while True:
#             r = await client.get(URL)
#             current_time = r.json()
#             await asyncio.sleep(1)

class TimeService:
    def __init__(self):
        self.current_time = None
        self.URL = "https://timeapi.io/api/Time/current/zone?timeZone=Europe/Moscow"

    async def start(self):
        async with httpx.AsyncClient() as client:
            while True:
                r = await client.get(self.URL)
                self.current_time = r.json()
                await asyncio.sleep(1)

    def get_time(self):
        return self.current_time

time_service = TimeService()